### PR TITLE
Preparations for release 1.0.0

### PR DESCRIPTION
**Pull request by ggalmazor**
This PR prepares this project towards the 1.0.0 release.

Review guide:
- The language level has been bumped to Java 11, which is the current LTS version
- Run the tests with `/.gradlew clean test`
- Build the client with `./gradlew clean build`
- When diff'ing with `master`, ignore imports and formatting
- Focus on groups of classes:
  - Data types on `com.dnsimple.data` have been made immutable POJOs with no coupling to third-party libraries.
  - Response types on `com.dnsimple.response` have been simplified into a set of container types (empty, simple, list, and paginated response) using generic types. 
    `ApiResponse` has lost the properties that coupled it to third-party libraries
  - Exception types on `com.dnsimple.exception` have been promoted to unchecked exceptions and follow the standard exception constructor standards.
    Now there are new exception types to represent the complete range of scenarios where the client can fail in a controlled fashion.
  - The endpoint classes at `com.dnsimple.endpoint` have been refactored to extract all the details about executing the HTTP requests.
    This has been abstracted away through the `HttpRequestFactory` interface.
    The previous implementation based on Google's API HTTP client library has been wiped out.
    There's an implementation of `HttpRequestFactory` based on the native Java11 `HttpClient` API at `com.dnsimple.endpoints.http.java11.Java11HttpRequestFactory`
    Users now can provide their own implementations of `HttpRequestFactory` and use them with our client if they wish to do so.

Pending/Backlog/Things that must go in v1 because doing them later would require bumping the version to v2:
- [ ] Improve Client instantiation to avoid having to rely on the global Dnsimple class
- [ ] Hook up a test coverage report to the build process
- [ ] Provide a StubClient that replays the test fixtures to make testing in integration easier for users.
- [ ] Review ASDF conf for the project
- [ ] Complete review of data types based on Go client baseline (and maybe c#)
- [ ] CI configuration
  - [x] Build
  - [x] Test 
  - [ ] Static type check
  - [ ] Coverage report
  - [ ] Other things?
- [ ] Maven repo publication
  - [ ] Publish snapshots / canary
  - [ ] Publish full releases
- [ ] Documentation review:
  - [ ] Update README.md, and CONTRIBUTING.md
  - [ ] Update CHANGELOG
  - [ ] Update developer site, provide new API examples, etc.
- [ ] Write templates for GitHub issues and PRs
- [ ] Automate refreshing the test fixture files with a Gradle task

### CODE REVIEW

**Comment by ggalmazor** [comment]
@ecomba, the PR requires more work, but I think I've iterated enough over the extraction of duplicated code. The classes at the core of that change are:
- https://github.com/dnsimple/dnsimple-java/blob/task/preparations_for_v1/src/main/java/com/dnsimple/endpoints/http/java11/Java11HttpRequestFactory.java 
- https://github.com/dnsimple/dnsimple-java/blob/task/preparations_for_v1/src/main/java/com/dnsimple/endpoints/http/java11/Java11ContainerResponseHandler.java
- https://github.com/dnsimple/dnsimple-java/blob/task/preparations_for_v1/src/main/java/com/dnsimple/endpoints/http/java11/Java11RawResponseHandler.java

The first two are the ones providing the higher kinded types that are used by all the Endpoint classes through the `HttpEndpointClient` class.

The primary goal for this change was to provide a single extension point for users of this library in case they want to provide their own HTTP client implementations (commonly through Apache Commons HTTP Client or Google API HTTP Client libs)

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1,2 @@
+gradle 6.5
+java adopt-openjdk-11+28
```
Comment (san983) [inline .tool-versions:2]:
```suggestion
java adoptopenjdk-11.0.0+28
```

**Comment by san983** [comment]
@ggalmazor qq, should `gradlew test` run the complete test suite?

**Comment by ggalmazor** [comment]
> @ggalmazor qq, should `gradlew test` run the complete test suite?

Yes, that'd be correct

File: .tool-versions (hunk 1 above)
Comment (ggalmazor) [reply .tool-versions:2]: Huh, is that what asdf suggested for you?

Comment (san983) [reply .tool-versions:2]: Right that's what appears for me on `asdf list-all java` as well for @ecomba. Perhaps you have a diff `asdf-java` plugin installed?

**Comment by san983** [comment]
> > @ggalmazor qq, should `gradlew test` run the complete test suite?
> 
> Yes, that'd be correct

Cool! I was confused since I haven't seen a summary in the output. But just realised it doesn't run it nothing changed. Just played a bit with the specs and realised that :)

**Comment by ggalmazor** [comment]
@san983, @ecomba, apologies. I think I can simplify the extension point at HttpEndpointClient to avoid having users deal with higher kinded types. I'm thinking that the most reasonable interface would output just an InputStream and then our client would transform that into our domain classes. I will study this and add some commits during the next days.

**Comment by ggalmazor** [comment]
Closing this PR in favor of a different release approach that's more incremental

**Pull request closed without merging**